Week-4/Exercise-4.1/site.js

```diff
@@ -0,0 +1,76 @@
+const getNumber = () => {
+  return Math.floor(Math.random() * 100);
+};
+
+function PromisePollyfill(executor) {
+  let onResolve,
+    onReject,
+    fulfilled = false,
+    rejected = false,
+    called = false,
+    value,
+    promiseResolutionRejectionTime;
+
+  function resolve(v) {
+    fulfilled = true;
+    value = v;
+    promiseResolutionRejectionTime = new Date();
+  }
+
+  function reject(reason) {
+    rejected = true;
+    value = reason;
+    promiseResolutionRejectionTime = new Date();
+  }
+
+  this.then = function (callback) {
+    onResolve = callback;
+
+    if (fulfilled && !called) {
+      called = true;
+      onResolve({ value, promiseResolutionRejectionTime });
+    }
+    return this;
+  };
+
+  this.catch = function (callback) {
+    onReject = callback;
+
+    if (rejected && !called) {
+      called = true;
+      onReject({ error: value, promiseResolutionRejectionTime });
```

Comment:
@AakashPandya Any reasons to send back resolutionRejectionTime ?

Reply:
> @AakashPandya Any reasons to send back resolutionRejectionTime ?

As per the assignment: **_Let’s also keep the promise resolution/rejection time as a variable_** 
I just thought to console log the timestamp when the promise got resolved/rejected !  @RajeshSivanesan